### PR TITLE
Start using uptane-sign and bump to latest version

### PR DESCRIPTION
**Pull request by EdTheBearded**
In order for us to be able to use `uptane-sign` we need some changes on `meta-updater`'s `classes/image_types_ostree.bbclass` and `classes/sota.bbclass`.  
On `classes/image_types_ostree.bbclass` I've changed the task `IMAGE_CMD:garagesign` to instead of using a hardcoded `garage-sign` to call `${GARAGE_SIGN_TOOL}`.
And `${GARAGE_SIGN_TOOL}` is being defined in `classes/sota.bbclass` with `garage-sign` as its default value. This made sense to me to not disrupt other users of this layer that might still be using `garage-sign`.
Then we go into `conf/distro/include/torizon.inc` and redefine `GARAGE_SIGN_TOOL` to `uptane-sign`.
These are the only changes made on those bbclasses, all else remains as is on `meta-updater`.

And on our Aktualizr recipe, I've bumped the version used to the latest (v3.2.4), and changed the variable names accordingly.

Related-to: TOR-3496

### CODE REVIEW

**Comment by EdTheBearded** [comment]
This change should be merged after our Aktualizr MR is closed (https://gitlab.int.toradex.com/rd/torizon-core/aktualizr/-/merge_requests/93).

@rborn-tx, do you think Phil could also review this change? Since I'd like to submit a PR to meta-updater, would be nice to have a review from him, like if I need to change something first.

**Comment by jsrc27** [comment]
@EdTheBearded, Just to make sure I understand. For now the plan is to copy these bbclasses into our layer and make the necessary adjustments for `uptane-sign`. Then in the future you plan to do these changes in upstream `meta-updater` right?

Did you already create a PR/thread discussing these potential changes in `meta-updater`?

**Comment by jsrc27** [comment]
> I haven't yet, I would like to see what Phil thinks about this, if he has any comments before opening a PR there.

@cajun-rat Your input and thoughts here would be appreciated.

**Comment by cajun-rat** [comment]
It would be much better if we can avoid two copies of the same code. Re-converging them once forked is always a massive pain.

Looking at the diff between the upstream one and this fork, making a backwards compatible version of `image_types_ostree.bbclass` in meta-updater seems be trivial: With a default value for `GARAGE_SIGN_TOOL` set to `garage-sign` (for now) it would appear to work for both. I don't see any problem approving the change in meta-updater, and that would avoid the fork.

File: classes/sota.bbclass
Comment (cajun-rat) [inline classes/sota.bbclass:38]: This is the only addition vs the meta-updater copy, we might as well upstream it directly

**Comment by rborn-tx** [comment]
I also think we shouldn't duplicate stuff from meta-updater here.

File: classes/sota.bbclass
Comment (EdTheBearded) [reply classes/sota.bbclass]: Not actually. I ended up creating a PR to meta updater, and there you can find the exact diff of the changes.
They're simple enough, just replacing the hardcoded tool name inside a task in classes/image_types_ostree.bbclass and then creating the tool variable name inside classes/sota.bbclass.

https://github.com/uptane/meta-updater/pull/104

**Comment by EdTheBearded** [comment]
I've changed this whole PR to simply modify the `GARAGE_SIGN_TOOL` name in our configs.
The actual change to enable bumping to more recent version was pushed to meta-updater:
https://github.com/uptane/meta-updater/pull/104

**Comment by jsrc27** [comment]
Okay, so we should wait for the PR on meta-updater to be reviewed and merged now right?

**Comment by EdTheBearded** [comment]
> Okay, so we should wait for the PR on meta-updater to be reviewed and merged now right?

Yeah, we should wait for meta-updater first (which won't break anything), then we should merge this PR and our internal aktualizr MR together, if possible.

File: classes/sota.bbclass
Comment (cajun-rat) [reply classes/sota.bbclass]: Approved, thank you!

**Comment by jsrc27** [comment]
Okay I see the PR in `meta-updater` has been merged now.

@rborn-tx @cajun-rat Any final comments from anyone? Or are we good to merge here?

**Comment by cajun-rat** [comment]
I'm happy. Thank you for your change to meta-updater!

**Comment by EdTheBearded** [comment]
> Okay I see the PR in `meta-updater` has been merged now.
> 
> @rborn-tx @cajun-rat Any final comments from anyone? Or are we good to merge here?

Hey @jsrc27! I don't think we can merge this yet, since the changes for aktualizr to be able to use `uptane-sign` weren't merged yet.

**Comment by jsrc27** [comment]
> I don't think we can merge this yet, since the changes for aktualizr to be able to use uptane-sign weren't merged yet.

Ahh sorry, I must have gotten the order these PRs need to be merged mixed up in my head.

**Comment by EdTheBearded** [comment]
Now that Aktualizr changes have finally been merged, I think we can go ahead with this PR.

**Comment by jsrc27** [comment]
> Now that Aktualizr changes have finally been merged, I think we can go ahead with this PR.

Actually the internal PR just merged to the `toradex-develop` branch on Aktualizr, it probably needs to then be merged to `toradex-master` for this to work properly in Yocto.

**Comment by EdTheBearded** [comment]
> > Now that Aktualizr changes have finally been merged, I think we can go ahead with this PR.
> 
> Actually the internal PR just merged to the `toradex-develop` branch on Aktualizr, it probably needs to then be merged to `toradex-master` for this to work properly in Yocto.

Oh, you're right! I just can't wait to close this one! :grinning:

**Review by jsrc27 — APPROVED** [review]

**Pull request merged**
